linera-chain/src/chain.rs

```diff
@@ -437,21 +437,6 @@ where
         Ok(true)
     }
 
-    /// Returns true if there are no more outgoing messages in flight up to the given
-    /// block height.
-    pub fn all_messages_delivered_up_to(&mut self, height: BlockHeight) -> bool {
```

Comment:
nit: Can we keep this function as it is here? (until we decide what to do with `outbox_counters`)

Reply:
The `outbox_counters` work great for validators but create a memory leak for clients with tracked chains.